backend/editor/api.py

```diff
@@ -41,16 +44,26 @@
 @app.on_event("startup")
 async def startup():
     """
-    Initialize database
+    Initialize Neo4J database
     """
-    initialize_db()
+    global driver, session
+    uri = settings.uri
+    driver = GraphDatabase.driver(uri)
 
 @app.on_event("shutdown")
 async def shutdown():
     """
-    Shutdown database
+    Close driver of Neo4J database
     """
-    shutdown_db()
+    driver.close()
+
+@app.middleware("http")
+async def transaction_creation(request: Request, call_next):
+    with driver.session() as session:
+        global txn
+        txn = session.begin_transaction()
+        response = await call_next(request)
+        return response

```

Comment:
You really have to use the context manager ! because here, you never end the transaction !
See https://neo4j.com/docs/api/python-driver/current/api.html#neo4j.Transaction

```suggestion
        with session.begin_transaction() as _txn:
            # make it current transaction
            txn = _txn
            response = await call_next(request)
        txn = None
        return response
```